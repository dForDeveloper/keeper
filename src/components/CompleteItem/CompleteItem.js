import React from 'react';
import PropTypes from 'prop-types';
import deleteicon from '../../images/deleteicon.svg';
import checkedicon from '../../images/checkedicon.svg';

const CompleteItem = (props) => {
  const { listItems, id, description } = props;
  return (
    <span key={id} className='NoteForm--span--complete'>
      <img
        src={checkedicon}
        className='NoteForm--icon--checked'
        onClick={() => props.handleComplete(listItems, id)}
        alt='checked icon'
      />
      <p className='NoteForm--p--complete'>{description}</p>
      <img
        src={deleteicon}
        className='NoteForm--icon--delete'
        onClick={() => props.handleItemDelete(listItems, id)}
        alt='delete icon'
      />
    </span>
  )
}

CompleteItem.propTypes = {
  listItems: PropTypes.array,
  id: PropTypes.string,
  description: PropTypes.string,
  handleComplete: PropTypes.func,
  handleItemDelete: PropTypes.func
}

export default CompleteItem;