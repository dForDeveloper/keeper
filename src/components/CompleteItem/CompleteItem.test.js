import React from 'react';
import { shallow } from 'enzyme';
import CompleteItem from './CompleteItem';

describe('CompleteItem', () => {
  let wrapper;
  const mockProps = {
    id: 'eia',
    description: 'Get groceries',
    handleComplete: jest.fn(),
    handleItemDelete: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(
      <CompleteItem {...mockProps} />
    );
  });

  it('should pass snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleComplete passing in the id', () => {
    wrapper.find('.NoteForm--icon--checked').simulate('click');
    expect(mockProps.handleComplete).toHaveBeenCalledWith('eia');
  });

  it('should call handleItemDelete passing in the id', () => {
    wrapper.find('.NoteForm--icon--delete').simulate('click');
    expect(mockProps.handleItemDelete).toHaveBeenCalledWith('eia');
  });
});