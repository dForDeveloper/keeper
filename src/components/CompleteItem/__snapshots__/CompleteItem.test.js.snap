// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompleteItem should pass snapshot test 1`] = `
<span
  className="NoteForm--span--complete"
>
  <img
    alt="checked icon"
    className="NoteForm--icon--checked"
    onClick={[Function]}
    src="checkedicon.svg"
  />
  <p
    className="NoteForm--p--complete"
  >
    Get groceries
  </p>
  <img
    alt="delete icon"
    className="NoteForm--icon--delete"
    onClick={[Function]}
    src="deleteicon.svg"
  />
</span>
`;
